app/imports/api/methods.js

```diff
@@ -10,21 +10,19 @@ import { FormSchema } from '/imports/schemas/TestForm.js';
 export const createApplication = new ValidatedMethod({
   name: 'application.create',
 
-  validate: new SimpleSchema({
-    userId: { type: String }
-  }).validator(),
+  // This doesn't receive any arguments, so we don't need validation
+  validate: null,
 
-  run({ userId }) {
+  run() {
 
     if( !this.userId ) {
       throw new Meteor.Error('Applications.methods.insert.not-logged-in', 'Must be logged in to create an application.');
     }
 
-    if( this.userId != userId ) {
-      throw new Meteor.Error('Applications.methods.insert.different-user', 'Who you trying to mess with, uh?');
-    }
+    Applications.insert({
+      userId: this.userId,
+    });
```

Comment:
what will happen after this executes and the Application exists? Will it run again thru this whole thing?